ivy.el

```diff
@@ -3671,6 +3671,12 @@ BUFFER may be a string or nil."
        (setq ivy--index 0)
        (ivy--reset-state ivy-last))
     "kill")
+   ("n"
+    ,(lambda (x)
+       (insert-and-inherit "#<buffer "
+                           (if (stringp x) x (car x))
+                           ">"))
+    "insert buffer name")
```

Comment:
I have a few comments/questions:

* What are the benefits of using `insert-and-inherit` instead of plain `inherit`? Should other Ivy/Counsel functions switch to `insert-and-inherit`? If there isn't a good reason to, I'd argue it's best to stay consistent with other insertion actions.
* I don't think `x` will ever be a list here, otherwise other action functions would break.
* Better use a standard [output function](https://www.gnu.org/software/emacs/manual/html_node/elisp/Output-Functions.html) instead of manual formatting.
* "insert buffer name" is misleading, because you are inserting its [printed hash notation](https://www.gnu.org/software/emacs/manual/html_node/elisp/Buffer-Type.html), not its [buffer name](https://www.gnu.org/software/emacs/manual/html_node/elisp/Buffer-Names.html). Because of this, I think the "n" key should be changed as well.

Given the above, I think you can simplify the function as

```el
(lambda (x)
  (prin1 (get-buffer x)))
```

or, if `insert-and-inherit` is deemed essential,

```el
(lambda (x)
  (insert-and-inherit (prin1-to-string (get-buffer x))))
```

Reply:
> Purely out of curiosity, what is the use-case for inserting a buffer's hash print notation, rather than plain name? Is this useful outside of eshell?

Well, one can find another usage, but I thought about eshell only.

Reply:
as for using insert-and-inherit, it preserves file name fontification if we have some